comtypes/test/test_clear_cache.py

```diff
@@ -0,0 +1,22 @@
+"""
+Test for the ``comtypes.clear_cache`` module.
+"""
+import io
+import runpy
+from unittest.mock import patch, call
+from unittest import TestCase
+
+from comtypes.client import _find_gen_dir
+
+
+class ClearCacheTestCase(TestCase):
+    # we patch sys.stdout so unittest doesn't show the print statements
+    @patch("sys.stdout", new=io.StringIO())
+    @patch("shutil.rmtree")
+    def test_clear_cache(self, mock_rmtree):
+        with patch("sys.argv", ["clear_cache.py", "-y"]):
```

Comment:
Very happy to make this change. I originally used a `with` for the `sys.argv` patch because I thought it was more obvious what it was doing when it was closer to the `runpy` call, but it is not exactly moving far away, and I agree that nesting `with` statements is untidy.